testproject/Assets/Tests/Runtime/NetworkSceneManager/NetworkObjectSceneMigrationTests.cs

```diff
@@ -0,0 +1,510 @@
+            for (int i = 0; i < 3; i++)
+            {
+                var serverInstance = Object.Instantiate(m_TestPrefabDestroyWithScene);
+                var serverNetworkObject = serverInstance.GetComponent<NetworkObject>();
+                // This set of NetworkObjects will be used to verify that NetworkObjets
+                // spawned with DestroyWithScene == true will get destroyed when the scene
+                // is unloaded
+                serverNetworkObject.Spawn(true);
+                m_ServerSpawnedDestroyWithSceneInstances.Add(serverNetworkObject);
+            }
+
+            yield return WaitForConditionOrTimeOut(VerifyAllClientsSpawnedInstances);
+            AssertOnTimeout($"Timed out waiting for all clients to spawn {nameof(NetworkObject)}s!");
+
+            // Now load three scenes
+            m_ServerNetworkManager.SceneManager.OnSceneEvent += SceneManager_OnSceneEvent;
+            for (int i = 0; i < 3; i++)
+            {
+                m_ClientsLoadedScene = false;
+                m_CurrentSceneLoading = m_TestScenes[i];
+                var loadStatus = m_ServerNetworkManager.SceneManager.LoadScene(m_TestScenes[i], LoadSceneMode.Additive);
+                Assert.True(loadStatus == SceneEventProgressStatus.Started, $"Failed to start loading scene {m_CurrentSceneLoading}! Return status: {loadStatus}");
+                yield return WaitForConditionOrTimeOut(() => m_ClientsLoadedScene);
+                AssertOnTimeout($"Timed out waiting for all clients to load scene {m_CurrentSceneLoading}!");
+            }
+
+            // Migrate the instances that don't synchronize with active scene changes into the 3rd loaded scene
+            // (We are making sure these stay in the same scene they are migrated into)
+            for (int i = 3; i < m_ServerSpawnedPrefabInstances.Count; i++)
+            {
+                SceneManager.MoveGameObjectToScene(m_ServerSpawnedPrefabInstances[i].gameObject, m_ScenesLoaded[2]);
+            }
+
+            // Migrate the instances that don't synchronize with active scene changes and are destroyed with the
+            // scene unloading into the 3rd loaded scene
+            // (We are making sure these get destroyed when the scene is unloaded)
+            for (int i = 0; i < m_ServerSpawnedDestroyWithSceneInstances.Count; i++)
+            {
+                SceneManager.MoveGameObjectToScene(m_ServerSpawnedDestroyWithSceneInstances[i].gameObject, m_ScenesLoaded[2]);
+            }
+
+            // Make sure they migrated to the proper scene
+            yield return WaitForConditionOrTimeOut(VerifySpawnedObjectsMigrated);
+            AssertOnTimeout($"Timed out waiting for all clients to migrate all NetworkObjects into the appropriate scenes!");
+
+            // Now change the active scene
+            SceneManager.SetActiveScene(m_ScenesLoaded[1]);
+            // We have to do this
+            Object.DontDestroyOnLoad(m_TestPrefabAutoSynchActiveScene);
```

Comment:
Another integration test thing where we don't want to destroy the temporary prefab instance during an integration test.